config/initializers/paperclip.rb

```diff
@@ -88,6 +88,16 @@
 
   Paperclip::Attachment.default_options[:s3_headers]['X-Amz-Storage-Class'] = ENV['S3_STORAGE_CLASS'] if ENV.has_key?('S3_STORAGE_CLASS')
 
+  # Some S3 providers are not compatible with the checksum mode
+  # See https://github.com/mastodon/mastodon/issues/26394
+  if ENV['S3_DISABLE_CHECKSUM_MODE'] == 'true'
+    class Aws::S3::FileDownloader
+      def validate!
+        @params.delete(:checksum_mode)
+      end
```

Comment:
This is fine, if very hackish. If going with this, I would still call upstream's `validate!` method:
```suggestion
      def validate!
        @params.delete(:checksum_mode)
        super
      end
```

Ideally, I'd prefer if `aws-s3-sdk` provided a way to disable use of that header, though.

Reply:
I can open an issue on their side but I have little hope